micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/CacheMeterBinder.java

```diff
@@ -47,23 +47,17 @@ public CacheMeterBinder(Object cache, String cacheName, Iterable<Tag> tags) {
 
     @Override
     public final void bindTo(MeterRegistry registry) {
-        if (size() != null) {
-            Gauge.builder("cache.size", cache.get(),
-                    c -> {
-                        Long size = size();
-                        return size == null ? 0 : size;
-                    })
+        Long cacheSize = size();
```

Comment:
This introduces a bug, causing the initial cache size to be reported from this point forward.

Reply:
(facepalm) Thank you for catching this! Addressed